src/time/Waveform.hpp

```diff
@@ -39,6 +38,11 @@ class Waveform {
    */
   int numberOfSamples();
 
+  /**
+   * @brief returns number of valid samples in time stored by this waveform
+   */
+  int numberOfValidSamples();
```

Comment:
Why is this function needed? Currently, it's only used in tests if I checked correctly.

Reply:
You are right. We can again hide this function and make the class a friend or use the correct fixture. I'll take care of this.

Reply:
I took care of this. Let's first merge https://github.com/precice/precice/pull/1094. Then this PR becomes a bit smaller, since the fixture is already there.